### PR TITLE
remove needs_ci_run script

### PR DESCRIPTION
**Pull request by nilsnolde**
~~it's failing again because `requests` isn't installed. let's remove it completely to remove that headache. it's not even simplifying things in this script, so it's really just overhead.~~

instead I removed it, it's breaking all the time silently and has hardly any impact even when it's working..

### CODE REVIEW

File: scripts/needs_ci_run [hunk 1]
```diff
@@ -41,17 +43,22 @@ def last_build_successful():
         return False, None
 
     # jobs defined in .circleci/config.yml
-    JOBS = ['lint-build-debug', 'build-release', 'build-osx']
+    JOBS = ['lint-build-debug', 'build-release']
```
Comment (nilsnolde) [inline scripts/needs_ci_run:46]: don't ask me why.. in fact.. should I change the PR to remove the damn thing? so sick of it.. if there's smth not needing any CI for sure, I'd rather quickly cancel those jobs and force merge

**Comment by nilsnolde** [comment]
delete it..

**Comment by nilsnolde** [comment]
deleted the script instead

**Review by kevinkreiser — APPROVED** [review]
i hate precommit hooks but since circle doesnt support a list of paths that should skip ci but it does support skipping ci via the commit message we could make a custom commit hook that fails if:

1. the only changes happening are on files in paths marked unimportant for ci AND
2. if the commit message doesnt have the `[skip ci]` string in it

this would at least trigger a person to think oh yeah i should skip ci here. problem is it wo uld only help people who use precommit which is probably not everyone.

i personally like it simple with no trickery regarding should or shouldnt build a commit

**Comment by nilsnolde** [comment]
That’d be fine too. I once customized a got command to “git draft” which automatically prepended “skip ci” but I never remembered to use it. It’s more transparent than a commit hook, but we can’t do that for people, they’ll have to do that. A quick and simple commit hook would go a long way for myself at least.

**Pull request merged**